src/App.tsx

```diff
@@ -1,24 +1,277 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useEffect, useMemo, useRef, useState,
+} from 'react';
 
-const USER_ID = 0;
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import * as todoService from './api/todos';
+import { Status } from './types/Status';
+import { USER_ID } from './api/userId';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [title, setTitle] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [status, setStatus] = useState<Status>(Status.All);
+  const [visible, setVisible] = useState(false);
+  const [disableInput, setDisableInput] = useState(false);
+  const [tempoTodo, setTempoTodo] = useState<Todo | null>(null);
+  const [isCompliteDeleting, setIsCompliteDeleting] = useState(false);
+
+  const completedTodosCount = todos.reduce((acc, todo) => {
+    return todo.completed ? acc + 1 : acc;
+  }, 0);
+
+  const todosCompleted = useMemo(() => todos.filter(
+    todo => todo.completed,
+  ), [todos]);
+
+  const activeTodosCount = todos.length - completedTodosCount;
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const visibleTodos = todos.filter(todo => {
+    switch (status) {
+      case Status.Active:
+        return !todo.completed;
+
+      case Status.Completed:
+        return todo.completed;
+
+      case Status.All:
+      default:
+        return true;
+    }
+  });
+
+  const resetAddTodoState = () => {
+    setTempoTodo(null);
+    setTitle('');
+    setDisableInput(false);
+  };
+
+  const createTempTodo = () => ({
+    id: 0,
+    userId: USER_ID,
+    title: title.trim(),
+    completed: false,
+  });
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const trimmedTitle = title.trim();
+
+    if (trimmedTitle === '') {
+      setErrorMessage('Title should not be empty');
```

Comment:
check all cases when you set an error and reset it after 3 seconds
it should fix such tests
better to make function which will take an error, set it and reset after 3 second and use in all places
![Screenshot from 2023-09-24 16-00-55](https://github.com/mate-academy/react_todo-app-add-and-delete/assets/33029864/655b3306-4e50-472d-9fca-9b33aba70c71)